Automattic/wp-calypso
# JPC: Clean redux state after the flow is completed

This PR adds a new action to empty the JPC redux subtree after a flow is completed. Until now, we were storing all kind of connection-related data that in a ver few cases created some problems with future connections down the line. Now we are cleaning all those not-needed-anymore subtrees once the user finish the flow.

How to test
=========
0. You need an unconnected site
1. Start from calypso.localhost:3000/jetpack/connect and connect your site (also you could start from wp-admin, but you would need to hack your wp.com sandbox or your jetpack plugin to make it work with your local calypso)
2. Connect the site. Once you pass the plans page (both purchasing a plan or not), the `jetpactConnect` redux subtrees should be empty

@roccotripaldi  @tyxla
